### PR TITLE
[Keymap] Add KC_MAKE keycode to my userspace and keymaps

### PR DESCRIPTION
**Pull request by stanrc85**
## Description
Add KC_MAKE keycode to my userspace and keymaps

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: users/stanrc85/stanrc85.h [hunk 1]
```diff
@@ -1,12 +1,19 @@
 #pragma once
 
 #include "quantum.h"
+#include "action.h"
```
Comment (drashna) [inline users/stanrc85/stanrc85.h]: This shouldn't actually be needed. 

```suggestion
```

File: users/stanrc85/stanrc85.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#include "stanrc85.h"
+
+__attribute__ ((weak))
+bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+  case KC_MAKE:
```
Comment (drashna) [inline users/stanrc85/stanrc85.c]: Actually, I've improved this a lot, so you may want to check out my userspace... and more specifically the `drashna_keymaps` branch on my fork.  It as all the goodies. :)

```c
  case KC_MAKE:  // Compiles the firmware, and adds the flash command based on keyboard bootloader
    if (!record->event.pressed) {
      #if !defined(KEYBOARD_viterbi)
        uint8_t temp_mod = get_mods();
        uint8_t temp_osm = get_oneshot_mods();
        clear_mods(); clear_oneshot_mods();
      #endif
      send_string_with_delay_P(PSTR("make " QMK_KEYBOARD ":" QMK_KEYMAP), 10);
        if (temp_mod & MODS_SHIFT_MASK || temp_osm & MODS_SHIFT_MASK ) {
          #if defined(__arm__)
            send_string_with_delay_P(PSTR(":dfu-util"), 10);
          #elif defined(BOOTLOADER_DFU)
            send_string_with_delay_P(PSTR(":dfu"), 10);
          #elif defined(BOOTLOADER_HALFKAY)
            send_string_with_delay_P(PSTR(":teensy"), 10);
          #elif defined(BOOTLOADER_CATERINA)
            send_string_with_delay_P(PSTR(":avrdude"), 10);
          #endif // bootloader options
        }
        if (temp_mod & MODS_CTRL_MASK || temp_osm & MODS_CTRL_MASK) { send_string_with_delay_P(PSTR(" -j8 --output-sync"), 10); }
        send_string_with_delay_P(PSTR(SS_TAP(X_ENTER)), 10);
        set_mods(temp_mod);
    }
    break;
```

This runs just the command to compile it, unless you hit/hold shift, first.  This is useful for testing.  
And if you hold control, it adds some additional commands to make it compile faster (useful on Windows)

But this is optional.

File: users/stanrc85/stanrc85.h (hunk 1 above)
Comment (stanrc85) [reply users/stanrc85/stanrc85.h]: Fixed!

File: users/stanrc85/stanrc85.c (hunk 2 above)
Comment (stanrc85) [reply users/stanrc85/stanrc85.c]: Very clever thanks!

**Comment by stanrc85** [comment]
Please don't merge yet, last commit with RESET didn't work as intended.

File: users/stanrc85/stanrc85.c (hunk 2 above)
Comment (drashna) [reply users/stanrc85/stanrc85.c]: Thanks, I've sent a lot of time tinkering with this :)

File: users/stanrc85/stanrc85.h [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
 #define LAYER1 1   //Default ANSI (enable with Fn2+CAPS)
 #define LAYER2 2   //Function keys, arrows, custom shortcuts, volume control
 #define LAYER3 3   //RGB Underglow controls and RESET
 
+enum custom_keycodes {
+  KC_MAKE = SAFE_RANGE,
+  NEW_SAFE_RANGE  //use "NEW_SAFE_RANGE" for keymap specific codes
+};
+
+#define MODS_SHIFT_MASK  (MOD_BIT(KC_LSHIFT)|MOD_BIT(KC_RSHIFT))
```
Comment (drashna) [inline users/stanrc85/stanrc85.h]: haha, I should probably update my code too, but these aren't needed anymore.  `MOD_MASK_*` is a core thing now, but this is fine, for now.

**Comment by drashna** [comment]
Thanks!

File: users/stanrc85/stanrc85.h (hunk 3 above)
Comment (stanrc85) [reply users/stanrc85/stanrc85.h]: Really? I got errors about them before I added here. Maybe I missed an include.

**Pull request merged**